src/authentication/realtime-authorization-spec.ts

```diff
@@ -0,0 +1,141 @@
+import { expect, use } from 'chai';
+import { Server } from 'hapi';
+import 'reflect-metadata';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+use(sinonChai);
+
+import { bootstrap } from '../config';
+import { getSignedAccessToken } from '../config/config-test';
+import { RealtimeHapiPlugin } from '../realtime';
+import { getTestServer } from '../server/hapi';
+import { MethodStubber, stubber } from '../shared/test';
+import TokenGenerator from '../shared/token-generator';
+import AuthenticationHapiPlugin from './authentication-hapi-plugin';
+import { generateTeamToken } from './team-token';
+
+const validAccessToken = getSignedAccessToken('idp|12345678', generateTeamToken(), 'foo@bar.com');
+async function getServer(
+  authenticationStubber: MethodStubber<AuthenticationHapiPlugin>,
+) {
+  const kernel = bootstrap('test');
+  kernel.rebind(AuthenticationHapiPlugin.injectSymbol).to(AuthenticationHapiPlugin);
+  kernel.rebind(RealtimeHapiPlugin.injectSymbol).to(RealtimeHapiPlugin);
+  const plugin = stubber<RealtimeHapiPlugin>(
+    p => sinon.stub(p, p.deploymentHandler.name)
+        .yields(200)
+        .returns(Promise.resolve(true)),
+    RealtimeHapiPlugin.injectSymbol,
+    kernel,
+  );
+
+  const authenticationPlugin = stubber(authenticationStubber, AuthenticationHapiPlugin.injectSymbol, kernel);
+  const server = await getTestServer(true, authenticationPlugin.instance, plugin.instance);
+  const tokenGenerator = kernel.get<TokenGenerator>(TokenGenerator.injectSymbol);
+
+  return {
+    server,
+    authentication: authenticationPlugin.instance,
+    tokenGenerator,
+  };
+}
+
+function arrange(
+  hasAccessToProject: boolean,
+  isAdmin = false,
+  isOpenDeployment = false,
+) {
+  return getServer(
+    (p: AuthenticationHapiPlugin) => {
+      return [
+        sinon.stub(p, p.userHasAccessToProject.name)
+          .returns(Promise.resolve(hasAccessToProject)),
+        sinon.stub(p, p.isAdmin.name)
+          .returns(Promise.resolve(isAdmin)),
+        sinon.stub(p, p.isOpenDeployment.name)
+          .returns(Promise.resolve(isOpenDeployment)),
+        sinon.stub(p, p.getProjectTeam.name)
+          .returns(Promise.resolve({id: 1, name: 'foo'})),
+      ];
+    },
+  );
+}
+
+describe('authorization for deployment events', () => {
+  describe('authenticated user', () => {
+    function makeRequest(server: Server, token: string) {
+      return server.inject({
+        method: 'GET',
+        url: `/events/deployment/1-1/${token}?token=${validAccessToken}`,
```

Comment:
What about the `/events/{teamId}` endpoint? Should tests be made for that as well?